src/shared.rs

```diff
@@ -72,4 +76,73 @@ pub async fn get_nonce_with_retry(
             }
         }
     }
+}
+
+/// @notice Burns a specific nonce by sending a 0 ETH transaction to self with a high gas price.
+/// @dev This function is used to invalidate a nonce by creating a dummy transaction.
+/// @param provider The Ethereum provider used to send the transaction.
+/// @param wallet The wallet used to sign the transaction.
+/// @param address The address whose nonce will be burned.
+/// @param nonce The specific nonce to burn.
+/// @param order_hash A string identifier for logging and tracing purposes.
+/// @return Returns Ok(()) if the transaction is sent and confirmed, or an error otherwise.
+pub async fn burn_nonce(
+    provider: &Arc<DynProvider<AnyNetwork>>,
+    wallet: &EthereumWallet,
+    address: Address,
+    nonce: u64,
+    order_hash: &str,
+) -> Result<(), anyhow::Error> {
+    let base_fee = provider
+        .get_gas_price()
+        .await?;
+
+    // Create a dummy transaction that sends 0 ETH to self with high gas price
+    let tx_request = WithOtherFields::new(TransactionRequest {
+        from: Some(address),
+        to: Some(address.into()),
+        value: Some(U256::ZERO),
+        nonce: Some(nonce),
+        gas: Some(ETH_TRANSFER_GAS), // Standard ETH transfer gas
+        gas_price: Some(base_fee * NONCE_BURN_GAS_MULTIPLIER),
+        max_fee_per_gas: Some(base_fee * NONCE_BURN_GAS_MULTIPLIER),
+        max_priority_fee_per_gas: Some(NONCE_BURN_PRIORITY_FEE),
```

Comment:
I guess the assumption here is that all fill txs would use a lower priority fee than this? did you do some math to arrive at this number?

Reply:
I believe the sequencer looks at total gas for replacement tx. Since I'm bidding 10x the base fee, it should be prioritized over the old tx. I just set the priority fee to some reasonable amount to make sure it's not excluded for non-zero priority fee.

Reply:
Actually, according to this article from MM, they recommend setting the priority bid 10% higher to override the old tx:
https://support.metamask.io/manage-crypto/transactions/how-to-speed-up-or-cancel-a-pending-transaction/

Reply:
> the sequencer looks at total gas for replacement tx

wait really! I didn't know that 👍

Reply:
If that's the case, and we don't know what the old priority bid was, we can set it to be the max possible bid in a priority auction (1e7 = .01 gwei).

Reasoning:
`priority bid = mps_of_improvement * bid_bps / BPS`
`max bid_bps = 10,000`
`max priority bid = mps_of_improvement * 1`
`max mps_of_improvement = 1e7`